pom.xml

```diff
@@ -59,7 +59,7 @@
       <dependency>
         <groupId>org.jenkins-ci.plugins</groupId>
         <artifactId>cloudbees-folder</artifactId>
-        <version>4.0</version>
+        <version>5.16</version>
```

Comment:
```
Folders Plugin v5.16 failed to load.
 - Credentials Plugin v1.24 is older than required. To fix, install v2.1.0 or later.
```

Means you need to add a `<dependencyManagement>` section requesting `credentials` in (at least) `2.1.0`.

Normally these sorts of problems are caught during `mvn validate` when using a sufficiently new parent POM, but there are some edge cases such as use of `optional` dependencies (probably the case here) which break that. So to check effectiveness, use

    mvn test -Dtest=InjectedTest